addons/mail/models/discuss/res_partner.py

```diff
@@ -77,11 +77,29 @@ def get_mention_suggestions_from_channel(self, channel_id, search, limit=8):
                 [("channel_ids", "in", channel.id)],
             ]
         )
-        partners = self._search_mention_suggestions(domain, limit)
+        extra_domain = expression.AND([
+            [('user_ids', '!=', False)],
+            [('user_ids.active', '=', True)],
+            [('partner_share', '=', False)]
+        ])
+        if channel.group_public_id.id:
+            extra_domain = expression.AND(
+                [
+                    extra_domain,
+                    [("user_ids.groups_id", "in", channel.group_public_id.id)],
+                ]
+            )
+        partners = self._search_mention_suggestions(domain, limit, extra_domain)
         members = self.env["discuss.channel.member"].search(
             [
                 ("channel_id", "=", channel.id),
                 ("partner_id", "in", partners.ids),
             ]
         )
-        return Store(members, fields={"channel": [], "persona": []}).add(partners).get_result()
+        store = Store(members, fields={"channel": [], "persona": []})
+        store.add(channel, {"authorizedGroupId": channel.group_public_id.id or None})
+        for p in partners:
+            store.add(p,{
+                "group_id": next((group.id for group in p.user_ids.groups_id if group.id == channel.group_public_id.id), None)
+            })
+        return store.get_result()
```

Comment:
👍 Ok for just passing manually group ids of partners that have the group_id of channel, for some privacy concerns.

Reply:
A partner is not in a single group, it can have many.

We don't have a UserModel in JS, this is all on persona, so ok to keep the shortcut on putting on partner/persona as field. Ok also to not be in `_to_store` as this is a specific data available when calling this method.

I would name it `groups_id`, and would define it as a `Record.many()` relational field in JS. You can use `[('ADD', data)]` to append a record to model. I would also introduce `ResGroups` in JS models, so that this is a model and the fields you introduce (`Persona/groups_id` and `Thread.group_public_id`) are relational fields (`Record.many()` and `Record.one()` respectively).

Doing so makes code rely more on models, JS models try to match as much as it can from python, and there's less custom code to maintain (we put more trust on good modeling and scoping the showing of model data).